src/common/bots/Bot.cpp

```diff
@@ -6,10 +6,23 @@ Bot::Bot(){};
  * Once it has enough of a resource, it will move to the next closest mine with a resource it needs.
  */
 std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
-Bot::getNearestGreedyMove(std::vector<std::shared_ptr<Mine>> mines,
-                          std::array<uint32_t, RESOURCE_TYPE_COUNT> current_resources,
-                          sf::Vector2f bot_center,
-                          std::unordered_map<uint32_t, uint32_t> mine_id_to_resource_count) {
+Bot::getNearestGreedyMove(uint32_t bot_player_id, GameObjectController& goc) {
```

Comment:
crazy how much computation this does per frame

Reply:
yeah, it feels like the right decision that we did 1 process per bot for that reason. bots stills only take up 3.5% CPU on avg on my ultrabook